create-deploy-pull-request/src/branch.ts

```diff
@@ -0,0 +1,47 @@
+import * as core from '@actions/core'
+import { GitHub } from '@actions/github/lib/utils'
+import { RequestError } from '@octokit/request-error'
+
+type Octokit = InstanceType<typeof GitHub>
+
+type CheckIfBranchExistsOptions = {
+  owner: string
+  repo: string
+  branch: string
+}
+
+export const checkIfBranchExists = async (octokit: Octokit, options: CheckIfBranchExistsOptions): Promise<boolean> => {
+  try {
+    await octokit.rest.repos.getBranch(options)
+    return true
+  } catch (error) {
+    if (error instanceof RequestError && error.status === 404) {
+      return false
+    }
+    throw error
+  }
+}
+
+type CreateUpdateBranchOptions = {
+  owner: string
+  repo: string
+  fromBranch: string
+  toBranch: string
+}
+
+export const createBranch = async (octokit: Octokit, options: CreateUpdateBranchOptions) => {
+  core.info(`Getting the commit of branch ${options.fromBranch}`)
+  const { data: fromBranch } = await octokit.rest.repos.getBranch({
+    owner: options.owner,
+    repo: options.repo,
+    branch: options.fromBranch,
+  })
+
+  core.info(`Creating branch ${options.toBranch} from ${fromBranch.commit.sha}`)
```

Comment:
`Creating a pull request from ${options.head} to ${options.base}`
と語順をあわせると更にわかりやすくなりそうです
`Creating a branch from ${fromBranch.commit.sha} to ${options.toBranch} `

Reply:
分けた方がわかりやすそうなので、以下のようにしてみます

```suggestion
  core.info(`From commit ${fromBranch.commit.sha} of branch ${options.fromBranch}`)
  core.info(`Creating a new branch ${options.toBranch}`)
```